pkg/proto/configuration/grpc/grpc.proto

```diff
@@ -365,6 +366,13 @@ message AuthenticationPolicy {
     // LOCAL_PEERCRED is used, which is typically capable of returning
     // up to 16 group memberships.
     string peer_credentials_jmespath_expression = 6;
+
+    // Forwards the authentication task to a remote server. This can be a
+    // convenient way to integrate custom authentication processes, for example
+    // if a proxy has already performed authentication where the resulting HTTP
+    // request's headers need to be verified and processed in a custom way to
+    // produce buildbarn.auth.AuthenticationMetadata.
+    RemoteGrpcRequestAuthenticationPolicy remote = 8;
```

Comment:
Would it make sense to replace this field and the `jwt` with one called `request_headers` or something, containing a `oneof` for `remote` and `jwt` inside? That way people can add new request header based authenticators without requiring further code changes to HTTP and gRPC.

Reply:
Looking at the factory code, it makes sense.